pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/utils/SchemaValidationFunctions.kt

```diff
@@ -0,0 +1,399 @@
+package gov.cdc.ocio.processingstatusapi.utils
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.JsonNodeType
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+
+import com.networknt.schema.JsonSchema
+import com.networknt.schema.JsonSchemaFactory
+import com.networknt.schema.SpecVersion
+import com.networknt.schema.ValidationMessage
+import gov.cdc.ocio.processingstatusapi.ReportManager
+import gov.cdc.ocio.processingstatusapi.exceptions.BadRequestException
+import gov.cdc.ocio.processingstatusapi.models.reports.CreateReportSBMessage
+import gov.cdc.ocio.processingstatusapi.models.reports.MessageMetadata
+import gov.cdc.ocio.processingstatusapi.models.reports.Source
+import gov.cdc.ocio.processingstatusapi.models.reports.StageInfo
+import gov.cdc.ocio.processingstatusapi.utils.SchemaValidation.Companion.logger
+import gov.cdc.ocio.processingstatusapi.utils.SchemaValidation.Companion.reason
+
+import java.awt.datatransfer.MimeTypeParseException
+import java.io.File
+import java.time.Instant
+import java.util.*
+import javax.activation.MimeType
+
+/**
+ * Function checks if input string is a valid JSON
+ * @param jsonString to be checked
+ * @return boolean true if its valid JSON otherwise false
+ */
+fun isJsonValid(jsonString: String): Boolean {
```

Comment:
As a general good practice, we should strive to avoid writing functions that aren't part of a class.  For example, the `isJsonValid` could be an Extension of String.  See https://kotlinlang.org/docs/extensions.html.